UCLA Radio/UCLA Radio/menu/ShowViewController.swift

```diff
@@ -23,17 +24,83 @@ class ShowViewController: BaseViewController {
     @IBOutlet weak var djsLabel: UILabel!
     @IBOutlet weak var blurbLabel: UILabel!
     
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var contentView: UIView!
+    @IBOutlet weak var notificationsView: UIView!
+    @IBOutlet weak var notificationsSwitch: UISwitch!
+    
+    
+    @IBAction func notificationsToggled(_ sender: UISwitch) {
+        if let show = show {
+            if #available(iOS 10.0, *) {
+                let current = UNUserNotificationCenter.current()
+                if sender.isOn {
+                    let requestIdentifier = show.title
+                    
+                    let content = UNMutableNotificationContent()
+                    content.title = "UCLA Radio"
+                    content.subtitle = show.title + " is on in 15 minutes!"
+                    content.body = "Woah! These new notifications look amazing! Don’t you agree?"
+                    
+                    var notificationTime = show.time
+                    notificationTime.minute = -15
+                    
+                    
+                    let trigger = UNCalendarNotificationTrigger(dateMatching: notificationTime,
+                                                                repeats: true)
+                    
+                    let request = UNNotificationRequest(identifier: requestIdentifier,
+                                                        content: content,
+                                                        trigger: trigger)
+                    
+                    current.add(request)
+                } else {
+                    current.removePendingNotificationRequests(withIdentifiers: [show.title])
+                }
+                
+
+            } else {
+                
+                let notification = UILocalNotification()
+                
+                notification.alertBody = show.title + " is on in 15 minutes!!"
+                
+                let calendar = NSCalendar(calendarIdentifier: .gregorian)!
+
+                let notificationDate = calendar.date(byAdding: DateComponents(minute: -15), to: show.getNextDateOfShow())!
+                
+                
+                print("Notification Date: \(notificationDate)")
+                
+                notification.fireDate = notificationDate
+                notification.repeatInterval = .weekOfYear
+                notification.repeatCalendar = Calendar.current
+                notification.soundName = UILocalNotificationDefaultSoundName;
+                
+                if sender.isOn {
+                    UIApplication.shared.scheduleLocalNotification(notification)
+                } else {
+                    UIApplication.shared.cancelLocalNotification(notification)
+                }
+            }
+        }
```

Comment:
Awesome -- I like that it handles the correct OS version for notification types.. Let's move this logic to a new place so we have data separated from our view controllers. Want to create a `NotificationManager` or similar class with class methods doing these actions?

One more thing: I don't think this is how we delete notifications - want to check on that? For reference, http://stackoverflow.com/questions/6340664/delete-a-particular-local-notification